pkg/ebpf/events_derived.go

```diff
@@ -116,3 +122,68 @@ func deriveContainerRemoved(t *Tracee) deriveFn {
 		return trace.Event{}, false, nil
 	}
 }
+
+func deriveDetectHookedSyscall(t *Tracee) deriveFn {
+	return func(event trace.Event) (trace.Event, bool, error) {
+		syscallsAdresses, err := getEventArgUlongArrVal(&event, "syscalls_addresses")
+		if err != nil {
+			return trace.Event{}, false, fmt.Errorf("error parsing syscalls_numbers arg: %v", err)
+		}
+		hookedSyscallData := t.ParseHookedAddresses(syscallsAdresses)
+		de := event
+		de.EventID = int(DetectHookedSyscallsEventID)
+		de.EventName = "hooked_syscalls"
+		de.ReturnValue = 0
+		de.Args = []trace.Argument{
+			{ArgMeta: trace.ArgMeta{Name: "hooked_syscalls", Type: "hookedSyscallData[]"}, Value: hookedSyscallData},
+		}
+		de.ArgsNum = 1
+		return de, true, nil
+	}
+}
+
+func (t *Tracee) ParseHookedAddresses(addresses []uint64) []bufferdecoder.HookedSyscallData {
+	hookedSyscallData := make([]bufferdecoder.HookedSyscallData, 0, 0)
+	for idx, syscallsAdress := range addresses {
+		InTextSegment, err := t.kernelSymbols.TextSegmentContains(syscallsAdress)
+		if err != nil {
+			continue
+		}
+		if !InTextSegment {
+			hookingFunction := t.ParseSymbol(syscallsAdress)
+			arch := t.config.OsConfig.GetOSReleaseFieldValue(helpers.OS_ARCH)
+			var syscallNumber int32
+			if strings.Compare(arch, "x86_64") == 0 {
```

Comment:
Suggestion - you can check it outside the loop and initialize a map according to it and prevent all this block.